account_payment_terminal/static/src/js/views/payment_terminal_form/payment_terminal_form_controller.js

```diff
@@ -126,6 +126,7 @@ odoo.define("account_payment_terminal.OCAPaymentTerminalFormController", functio
             return this._super.apply(this, arguments).then(function () {
                 return self._get_payment_info().then(function (payment_info) {
                     self.oca_payment_terminal_id = payment_info.oca_payment_terminal_id;
+                    payment_info.terminal_id = payment_info.oca_payment_terminal_id;
```

Comment:
IMO, it can be done here

https://github.com/acsone/account-payment/blob/452edcb562786f1042bef457706ec3c3e69bf59f/account_payment_terminal/wizards/account_payment_register.py#L23-L35

Reply:
Done